dio/pubspec.yaml

```diff
@@ -9,20 +9,20 @@ description: |
 homepage: https://github.com/cfug/dio
 repository: https://github.com/cfug/dio/blob/main/dio
 issue_tracker: https://github.com/cfug/dio/issues
-version: 5.1.1
+version: 5.1.2
 
 environment:
   sdk: '>=2.15.0 <3.0.0'
 
 dependencies:
-  http_parser: ^4.0.0
+  http_parser: ^4.0.2
```

Comment:
There's no need to unnecessarily constrain this to the newest version. It would force downstream user to use the newest version even if we're still compatible with previous versions.

In general it's a good idea to constrain our dependencies as loosely as possible so that we don't cause dependency conflicts with other libraries or the consumer of this library.

If a user wants to use the latest http parser version, he can already do so. We don't need to update that on our side.

Constraining dev dependencies to the latest version makes sense, since we can make use of more lints etc and because they can't create dependency conflicts.

So please revert this change and the version number bump here.

Reply:
AFAIK dev dependencies do create conflicts, just thinking of Flutter and `flutter_test`. I think we should revert all the changes to dev dependencies or align them with the minimum supported Flutter version.

Reply:
but we don't depend on Flutter test for the changed projects?

Reply:
But if I want to use Dio in a Flutter project, this might raise the test package higher than supported by Flutter.

Reply:
Why would it? dev_dependencies aren't resolved by consumers of a particular library. See https://dart.dev/tools/pub/dependencies#dev-dependencies

Reply:
Hmm yeah, seems to bi that way. I remember having conflicts somehow. Nvm.